src/common/start_utils.nim

```diff
@@ -22,46 +22,53 @@ proc startCoreProcess*(traceId: int, recordCore: bool, callerPid: int, test: boo
     echo "warning: tried to create symlink to last, but error: ", e.msg
     echo "continuing despite that, you just won't have the special `last` symlink"
 
-  if IS_DB_BASED[trace.lang]:
-    startCustomBackend(trace.outputFolder, recordCore, callerPid, noOutput)
-  else:
-    echo "The specified recording is not compatible with the current version of CodeTracer"
-    quit 1
-
-proc startCustomBackend(traceFolder: string, recordCore: bool, callerPid: int, noOutput: bool = false): Process =
-  # echo "  custom backend path", dbBackendExe
+  let ct_config = loadConfig(folder=getCurrentDir(), inTest=false)
 
   let workdir = codetracerInstallDir
   let options: set[ProcessOption] = {poParentStreams}
-
+  let traceFolder = trace.outputFolder
   # TODO: not sure why noOutput true for
   # nix build: if not noOutput: {poParentStreams} else: {}
 
   putEnv("CODETRACER_LINKS_PATH", linksPath)
-  let process = if not recordCore:
+  if not recordCore:
     debugprint "not recordCore"
     debugprint "noOutput ", noOutput
     debugprint "options ", options
-
-    startProcess(
-      dbBackendExe,
-      workingDir = workdir,
+    if IS_DB_BASED[trace.lang]:
+      result = startProcess(
+        dbBackendExe,
+        workingDir = workdir,
         args = @[
           $callerPid,
           traceFolder / "trace.json",
           traceFolder / "trace_metadata.json"
         ],
         options=options)
-    else:
-      startProcess(
-        codetracerExe,
+    elif ct_config.rr_backend_enabled:
```

Comment:
can we unify the args/CLI api for all backends(db, rr etc): or is it something to be done in the future/in protocol?